notebooks/setting_up_netcdf_stitching.py

```diff
@@ -0,0 +1,139 @@
+
+    # TODO figure out why the date range is so does not include
+    expected_len = len(pd.date_range(start=start_yr+"-01-01", end=end_yr+"-12-31", freq='M'))
+    assert (len(dat) == expected_len), "Not enough data in " + file + "for period " + start_yr + "-" + end_yr
+
+    return dat
+
+
+def get_data_from_pangeo(fl):
+    """Download all of the data from a pangeo file list
+
+        :param fl:             list of the files to get data from
+        :return:               list of xarray
+    """
+    data_list = []
+
+    # this for loop is really really freaking slow need to figure out how
+    # the heck to do this also only want to have to run it once runing it mulitple
+    # times is awuful!
+    for f in fl:
+        o = load_data(f)
+        data_list.append(o)
+
+    return data_list
+
+
+def stitch_gridded(rp, dl, fl):
+    """stitch the gridded data together.
+
+        :param rp:             data frame of the recipes
+        :param dl:             list of the data files
+        :param fl:             list of the data file names
+        :return:               xarray data set
+
+        TODO add a check to make sure that there is only one stitching id being passed into
+        the function.
+    """
+
+    rp.reset_index(drop=True, inplace=True)
+    out = get_netcdf_values(i=0, dl=dl, rp=rp, fl=fl)
+
+    for i in range(1,len(rp)):
+        new_vals = get_netcdf_values(i=i, dl=dl, rp=rp, fl=fl)
+        out = np.concatenate((out, new_vals), axis=0)
+
+    # Create a time series with the target data information
+    # TODO this needs to be some sort of check that allows you to make sure that there is only 1 recipe being stitched
+    # select monthly, daily, or annual data.
+    start = str(min(rp["target_start_yr"]))
+    end = str(max(rp["target_end_yr"]))
+
+    # Note that the pd.date_range call need the date/month defined otherwise it will
+    # truncate the year from start of first year to start of end year which is not
+    # what we want. We want the full final year to be included in the times series.
+    times = pd.date_range(start=start+"-01-01", end=end+"-12-31", freq='M')
+    assert (len(out) == len(times)), "Problem with the length of time"
+
+    # Extract the lat and lon information that will be used to structure the
+    # empty netcdf file.
+    lat = dl[0].lat.values
+    lon = dl[0].lon.values
+
+    # Creat the xarray data set
+    rslt = xr.Dataset({'tas': xr.DataArray(
+        out,
+        coords=[times, lat, lon],
+        dims=["time", "lat", 'lon'],
+        # TODO what other meta data should be added? A path to the recepie file?
+        attrs={'units': 'K',
+               'stitching_id': rp['stitching_id'].unique()})
+    })
+
+    return rslt
+
+
+########################################################################################
+# Load the recipe for the gridded product developed from the R package.
+recipe = pd.read_csv('./stitches_dev/recipes_for_python.csv')
```

Comment:
I had to change the path from './notebooks/stitches_dev/recipes_for_python.csv' to this string instead because this .py file is in the notebooks directory already. Just fyi